funcx_sdk/funcx/sdk/executor.py

```diff
@@ -242,19 +244,25 @@ def _submit_task_kernel(self, kill_event: threading.Event):
         kill_event : threading.Event
             Sentinel event; used to stop the thread and exit.
         """
+        to_send = self.task_outgoing  # cache lookup
+        interval = self.batch_interval
         while not kill_event.is_set():
             tasks: t.List[TaskSubmissionInfo] = []
-            start = time.time()
             try:
-                while (
-                    time.time() - start < self.batch_interval
-                    and len(tasks) < self.batch_size
-                ):
-                    tasks.append(self.task_outgoing.get(timeout=0.1))
+                task = to_send.get()  # Block while waiting for first result ...
+                beg = time.time()
+                while task is not None:
+                    tasks.append(task)
+                    if (
+                        not (len(tasks) < self.batch_size)
+                        or time.time() - beg > interval
+                    ):
+                        break
+                    task = to_send.get(block=False)  # ... but don't block thereafter
             except queue.Empty:
                 pass
             if tasks:
-                log.info(f"Submitting {len(tasks)} tasks to funcX")
+                log.info(f"Submitting tasks to funcX: {len(tasks)}")
```

Comment:
I used to write log lines like this until I was told that log lines should read like sentences. This is subjective, but I think it makes logs friendlier.